ENH: Make print_thing respect display.precision for Real numbers

aydinomer00 opened the pull request:
- [x] closes #60503

- [x] Tests added and passed:
  - Added test_pprint_thing_real_precision to test precision formatting
  - Verified with multiple display.precision values (2, 3, 4)
  - Tested with various float numbers including np.pi

- [x] All code checks passed

- [x] Added type annotations for clarity in modified functions

- [x] Added an entry in the latest doc/source/whatsnew/vX.X.X.rst file:
      - Enhanced print_thing to respect display.precision setting for Real numbers

This PR modifies the print_thing function to respect display.precision settings when formatting Real numbers. For example:

```python
with pd.option_context('display.precision', 2):
    df = pd.DataFrame({"value": [np.pi]})
    print(df)  # Shows 3.14

with pd.option_context('display.precision', 4):
    print(df)  # Shows 3.1416

aydinomer00 committed:
ENH: Make print_thing respect display.precision for Real numbers

aydinomer00 committed:
ENH: Make print_thing respect display.precision for Real numbers (#60503)

rhshadrach commented on pandas/io/formats/printing.py:
```diff
@@ -178,42 +180,50 @@ def pprint_thing(
 
-        result = str(thing)
+        # Real instance kontrolü ve precision uygulaması
+        if isinstance(thing, Real) and not isinstance(thing, int):
```
Can you use `is_float` from `padnas.core.dtypes.inference`

rhshadrach commented on pandas/io/formats/printing.py:
```diff
@@ -178,42 +180,50 @@ def pprint_thing(
             escape_chars = escape_chars or ()
 
-        result = str(thing)
+        # Real instance kontrolü ve precision uygulaması
```
This comment repeats the code verbatim and can be removed. Also, can you make any comments in English.

rhshadrach commented on pandas/tests/io/formats/test_printing.py:
```diff
@@ -2,9 +2,12 @@
+import numpy as np
 import pytest
 
+from pandas._config.config import option_context  # option_context
```
Can you remove the comment

rhshadrach commented on pandas/tests/io/formats/test_printing.py:
```diff
@@ -155,6 +158,13 @@ def test_east_asian_len(self):
         assert adj.len("パンダpanda") == 11
         assert adj.len("ﾊﾟﾝﾀﾞpanda") == 10
 
+    def test_pprint_thing_real_precision(self):
```
Can you move to TestPPrintThing above

rhshadrach commented on pandas/tests/io/formats/test_printing.py:
```diff
@@ -155,6 +158,13 @@ def test_east_asian_len(self):
 
+    def test_pprint_thing_real_precision(self):
+        with option_context('display.precision', 3):
+            assert pprint_thing(3.14159265359) == "3.142"
```
Use `printing.pprint_thing` instead, as the other tests here do

rhshadrach reviewed (changes requested):
Thanks for the PR!

aydinomer00 committed:
REF: Improve print_thing implementation and tests

- Replace isinstance(thing, Real) with is_float
- Remove Turkish comments and unnecessary inline comments
- Move test_pprint_thing_real_precision to TestPPrintThing class
- Use consistent printing.pprint_thing import style

aydinomer00 committed:
REF: Clean up print_thing implementation

- Remove Real numbers reference from docstring
- Remove Turkish comments
- Fix import structure in tests

rhshadrach commented:
@aydinomer00 - looks like the CI is failing. Let me know if you'd like assistance.

aydinomer00 committed:
Apply reviewer requests: use is_float, remove extra comments, etc.

rhshadrach commented:
Closing as stale. @aydinomer00 - if you'd like to continue, merge main and address the comments above, and we'd be more than happy to reopen!
